src/gui/updater/ocupdater.cpp

```diff
@@ -65,8 +65,16 @@ bool OCUpdater::performUpdate()
 
 void OCUpdater::backgroundCheckForUpdate()
 {
-    // FIXME
-    checkForUpdate();
+    int dlState = downloadState();
+
+     if( dlState == Unknown ||
+             dlState == UpToDate         ||
+             /* dlState == DownloadComplete ||  <- are we checking if a previous download was successful already? */
+             dlState == DownloadFailed   ||
+             dlState == DownloadTimedOut ) {
+         // how about  UpdateOnlyAvailableThroughSystem?
+         checkForUpdate();
```

Comment:
I'd prefer having this as a fully-enumerated switch statement so newly added members cause a warning/error because you're of course right with adding this safe-guard. This can be nasty.